k9mail/src/main/java/com/fsck/k9/mailstore/migrations/MigrationTo58.java

```diff
@@ -0,0 +1,31 @@
+package com.fsck.k9.mailstore.migrations;
+
+
+import android.database.sqlite.SQLiteDatabase;
+
+
+class MigrationTo58 {
+    static void cleanUpOrphanedData(SQLiteDatabase db) {
+        cleanUpFtsTable(db);
+        cleanUpMessagePartsTable(db);
+    }
+    
+    private static void cleanUpFtsTable(SQLiteDatabase db) {
+        MigrationTo56.cleanUpFtsTable(db);
+    }
+
+    private static void cleanUpMessagePartsTable(SQLiteDatabase db) {
+        db.execSQL("DELETE FROM message_parts WHERE root NOT IN " +
+                "(SELECT message_part_id FROM messages WHERE deleted = 0)");
```

Comment:
Why would `row IN (NULL)` always be true? That makes no logical sense to me. Doing a quick test doesn't support this either:

```
sqlite> CREATE TABLE a (a INT);
sqlite> INSERT INTO a VALUES (5);
sqlite> SELECT * FROM a WHERE a IN (NULL);
sqlite> SELECT * FROM a WHERE a NOT IN (NULL);
sqlite> SELECT * FROM a WHERE a IN (5);
5
sqlite> SELECT * FROM a WHERE a NOT IN (5);
sqlite> SELECT * FROM a WHERE a IN (5, NULL);
5
sqlite> SELECT * FROM a WHERE a NOT IN (5, NULL);
sqlite>
```

Am I missing something here? Please check again and explain if I got this wrong. Nice to know more people are looking at stuff and catching errors though :)